v3/src/components/graph/models/graph-content-model.ts

```diff
@@ -93,25 +93,25 @@ export const GraphContentModel = DataDisplayContentModel
       return self.plotType === 'scatterPlot' && ['left', 'bottom'].includes(place)
     },
     getUpdateCategoriesOptions(resetPoints=false): IUpdateCategoriesOptions {
-      const xAttrId = self.getAttributeID("x"),
-        xAttrType = self.config.attributeType("x"),
+      const xAttrId = this.getAttributeID("x"),
```

Comment:
Rather than using `this` here it would be preferable to put `getUpdateCategoriesOptions()` into its own `views()` block because `self` gives better typing than `this`. If you don't want to deal with it here I can deal with it on rebase of the next PR.

Reply:
Good idea